lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -0,0 +1,226 @@
+"""
+User Partitions Transformer
+"""
+from openedx.core.lib.block_cache.transformer import BlockStructureTransformer
+
+from .split_test import SplitTestTransformer
+from .utils import get_field_on_block
+
+
+class UserPartitionTransformer(BlockStructureTransformer):
+    """
+    A transformer that enforces the group access rules on course blocks,
+    by honoring their user_partitions and group_access fields, and
+    removing all blocks in the block structure to which the user does
+    not have group access.
+
+    Staff users are *not* exempted from user partition pathways.
+    """
+    VERSION = 1
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class;
+        same identifier used in setup.py.
+        """
+        return "user_partitions"
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Computes any information for each XBlock that's necessary to execute
+        this transformer's transform method.
+
+        Arguments:
+            block_structure (BlockStructureCollectedData)
+        """
+        # First have the split test transformer setup its group access data for each block.
+        SplitTestTransformer.collect(block_structure)
```

Comment:
Yes.  The `UserPartitionsTransformer` explicitly contains the `SplitTestTransformer`.  I explained that in the docstring of the `SplitTestTransformer`, but may not have in the docstring of the `UserPartitionsTransformer`.  I can update the docs.